### PR TITLE
Add bundle pre-upload

### PR DESCRIPTION
**Pull request by dgusakov**
Add ADCMInitializer class for ADCM image init operations and add preupload_bundle_urls option to allow pre-upload of bundles into ADCM before committing image

Refactor ADCM API Auth and ADCMClient creation

Remove deprecated fixtures

Add tests output to allure for plugin tests

### CODE REVIEW

**Comment by dgusakov** [comment]
bender build

File: src/adcm_pytest_plugin/fixtures.py [hunk 1]
```diff
@@ -52,10 +53,17 @@
 ]
 
 
+@allure.title("ADCM credentials")
+@pytest.fixture(scope="session")
+def adcm_credentials():
```
Comment (vgorkavenko) [inline src/adcm_pytest_plugin/fixtures.py:58]:
```suggestion
def adcm_credentials() -> dict:
```

**Review by vgorkavenko — APPROVED** [review]

File: src/adcm_pytest_plugin/fixtures.py (hunk 1 above)
Comment (dgusakov) [reply src/adcm_pytest_plugin/fixtures.py]: Done

**Review by vgorkavenko — DISMISSED** [review]

**Review by madlabman — DISMISSED** [review]

**Comment by dgusakov** [comment]
bender build

**Review by madlabman — APPROVED** [review]

**Review by vgorkavenko — APPROVED** [review]

**Pull request merged**